File: unit1/population.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt for start size
    int startSize;
    do
    {
        startSize = get_int("Start size: ");
    }
    while (startSize < 9);

    // Prompt for end size
    int endSize;
    do
    {
        endSize = get_int("End size: ");
    }
    while (endSize < startSize);

    // Calculate number of years until we reach threshold
    int numYears = 0;

    while (startSize < endSize)
    {
        int newLlamas = startSize / 3;
        int ripLlamas = startSize / 4;
        // startSize = startSize + newLlamas - ripLlamas;
        startSize += newLlamas - ripLlamas;                     // compound assignment
        // numYears = numYears + 1;
        // numYears += 1;
        numYears++;                                             // increment operator
        // printf("DEBUG - Updated population: %d\n", startSize);
    }

    // Print number of years
    printf("Years: %d\n", numYears);
}

File: unit3/population.py
import cs50

while True:
    start_size = cs50.get_int("Start size: ")
    if start_size >= 9:
        break

while True:
    end_size = cs50.get_int("End size: ")
    if end_size >= start_size:
        break

num_years = 0

while start_size < end_size:
    new_llamas = start_size // 3
    rip_llamas = start_size // 4
    start_size += new_llamas - rip_llamas
    num_years += 1
    # print(f"DEBUG - Updated population: {start_size}")

print(f"Years: {num_years}")